src/main/java/edu/kit/ipd/crowdcontrol/objectservice/moneytransfer/MoneyTransferManager.java

```diff
@@ -162,7 +173,7 @@ private void fetchNewGiftCodes() throws MoneyTransferException {
             try {
                 Optional<GiftCodeRecord> rec = MailParser.parseAmazonGiftCode(message, parsingPassword);
                 if (rec.isPresent()) {
-                    workerBalanceOperations.addGiftCode(rec.get().getCode(), rec.get().getAmount());
+                    workerBalanceOperations.addGiftCode(rec.get().getCode(), (int) (rec.get().getAmount() * eurUsdRate));
                     giftCodesCount++;
                 }
```

Comment:
no, because if it's not present, the mail is no giftcode mail (but e.g. Spam). If the mail is a giftcode mail and cannot be parsed, parseAmazonGiftCode() throws an exception